compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TilingInterfaceImpl.cpp

```diff
@@ -1686,18 +1745,31 @@ LogicalResult AttentionOp::getResultTilePosition(
     OpBuilder &builder, unsigned resultNumber, ArrayRef<OpFoldResult> offsets,
     ArrayRef<OpFoldResult> sizes, SmallVector<OpFoldResult> &resultOffsets,
     SmallVector<OpFoldResult> &resultSizes) {
-  if (resultNumber == 0) {
-    ArrayRef<int64_t> resultShape = getOutputType().getShape();
-    resultSizes = getAsOpFoldResult(builder.getIndexArrayAttr(resultShape));
-    resultOffsets =
-        SmallVector<OpFoldResult>(getOutputRank(), builder.getIndexAttr(0));
-    for (auto [idx, info] : llvm::enumerate(llvm::zip_equal(offsets, sizes))) {
-      resultOffsets[idx] = std::get<0>(info);
-      resultSizes[idx] = std::get<1>(info);
-    }
-    return success();
+  resultOffsets.clear();
+  resultSizes.clear();
+
+  AffineMap resultIndexingMap;
+  switch (resultNumber) {
+  case 0:
+    resultIndexingMap = getOutputMap();
+    break;
+  case 1:
+    resultIndexingMap = *getMaxMap();
+    break;
+  case 2:
+    resultIndexingMap = *getSumMap();
+    break;
+  default:
+    return failure();
   }
-  return failure();
+
+  for (AffineExpr dimExpr : resultIndexingMap.getResults()) {
+    AffineDimExpr dim = cast<AffineDimExpr>(dimExpr);
+    int64_t pos = dim.getPosition();
```

Comment:
How about we collapse them into a single statement? I.e., `int64_t pos = cast<...>(...).getPosition()`

Reply:
done